src/Microsoft.TestPlatform.CommunicationUtilities/TcpClientExtensions.cs

```diff
@@ -71,7 +71,11 @@ internal static class TcpClientExtensions
                         ioException,
                         remoteEndPoint,
                         localEndPoint);
-                    error = ioException;
+                    // Do not pass the error to the caller, the transport was torn down because testhost
```

Comment:
Not super happy about this, but if we do it later in the code we end up ignoring errors for all types of transports (we have just 1 type now, but the abstractions are there). Or we have to match a specific type of error, and then make sure it was raised here, which is pretty much the same as not raising it at all.